numba/np/ufunc/array_exprs.py

```diff
@@ -329,6 +329,28 @@ def _legalize_parameter_names(var_list):
             var.name = old_name
 
 
+class _EraseInvalidLineRanges(ast.NodeTransformer):
```

Comment:
What is this fixing? Is there an example? I ask this as I'm wondering if needing this is simply an effect of "strange" line number information being generated earlier on in the compilation pipeline?

Reply:
Anything with arrayexprs and a lot of stencil tests fail without this. This has nothing to do with line number information being wrong. It's completely caused by inlining and the way arrayexpr generate ASTs and splicing ASTs. In Py3.11, AST nodes contain start and end linenos. Arrayexpr splices AST in a way that can generate nodes with outer AST node saying the line range is (1, 10) and the inner nodes are at line 100. This fixes it.

Reply:
Ah I see. Thanks for the explanation. IIRC the `arrayexpr` node line numbers aren't quite right anyway, e.g. #7775.